src/lib.rs

```diff
@@ -262,20 +262,125 @@ impl<T> StableVec<T> {
     /// This method uses exactly the same deletion strategy as
     /// [`remove()`](#method.remove).
     ///
+    /// # Example
+    ///
+    /// ```
+    /// # use stable_vec::StableVec;
+    /// let mut sv = StableVec::from(&[1, 2, 3]);
+    /// assert_eq!(sv.pop(), Some(3));
+    /// assert_eq!(sv.into_vec(), vec![1, 2]);
+    /// ```
+    ///
     /// # Note
     ///
     /// This method needs to find index of the last valid element. Finding it
     /// has a worst case time complexity of O(n). If you already know the
     /// index, use [`remove()`](#method.remove) instead.
     pub fn pop(&mut self) -> Option<T> {
-        let last_index = self.deleted
-            .iter()
-            .enumerate()
-            .rev()
-            .find(|&(_, deleted)| !deleted)
-            .map(|(i, _)| i)
-            .unwrap_or(0);
-        self.remove(last_index)
+        self.find_last_index().and_then(|index| self.remove(index))
+    }
+
+    /// Removes and returns the first element from this collection, or `None` if
+    /// it's empty.
+    ///
+    /// This method uses exactly the same deletion strategy as
+    /// [`remove()`](#method.remove).
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use stable_vec::StableVec;
+    /// let mut sv = StableVec::from(&[1, 2, 3]);
+    /// assert_eq!(sv.pop_first(), Some(1));
+    /// assert_eq!(sv.into_vec(), vec![2, 3]);
+    /// ```
+    ///
+    /// # Note
+    ///
+    /// This method needs to find index of the first valid element. Finding it
+    /// has a worst case time complexity of O(n). If you already know the
+    /// index, use [`remove()`](#method.remove) instead.
+    pub fn pop_first(&mut self) -> Option<T> {
+        self.find_first_index().and_then(|index| self.remove(index))
+    }
+
+    /// Finds the first element and returns a reference to it, or `None` if it's empty.
+    ///
+    /// This method has O(n) complexity.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use stable_vec::StableVec;
+    /// let mut sv = StableVec::from(&[1, 2]);
+    /// sv.remove(0);
+    /// assert_eq!(sv.find_first(), Some(&2));
+    /// ```
+    pub fn find_first(&self) -> Option<&T> {
```

Comment:
Maybe just `first` (like for `Vec<T>`)? Or do you think `find_` is important to hint on the O(n) complexity? But `pop` and `pop_first` don't have such a hint in their names.

Reply:
That's roughly the reasoning behind it. I thought it is necessary to distinguish methods like this from those of std::vec::Vec, because the user might think that if it's called `first`, then it should behave the same as `Vec::first()` - after all, this is "stable `Vec`".

Reply:
Mh ok, yeah I'm fine with keeping it `find_first` for now.